test/unit-test/FreeRTOS_IPv6/FreeRTOS_IPv6_utest.c

```diff
@@ -0,0 +1,938 @@
+    NetworkEndPoint_t xEndpoint;
+    /* Different IPv6 address is 2001:1234:5678::FFFF */
+    IPv6_Address_t xDiffIPAddress = { 0x20, 0x01, 0x12, 0x34, 0x56, 0x78, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF };
+
+    memset( &xEndpoint, 0, sizeof( xEndpoint ) );
+    xEndpoint.bits.bIPv6 = 1U;
+    memcpy( xEndpoint.ipv6_settings.xIPAddress.ucBytes, xDiffIPAddress.ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+    pxNetworkBuffer->pxEndPoint = &xEndpoint;
+
+    FreeRTOS_FindEndPointOnIP_IPv6_ExpectAndReturn( &( pxTCPPacket->xIPHeader.xSourceAddress ), NULL );
+    FreeRTOS_IsNetworkUp_IgnoreAndReturn( 1 );
+
+    eResult = prvAllowIPPacketIPv6( &( pxTCPPacket->xIPHeader ), pxNetworkBuffer, 0U );
+    TEST_ASSERT_EQUAL( eReleaseBuffer, eResult );
+}
+
+/**
+ * @brief test_eHandleIPv6ExtensionHeaders_TCP_happy_path
+ * Prepare a packet have extension with following order. Check if eHandleIPv6ExtensionHeaders determines to process it.
+ *  - ipIPv6_EXT_HEADER_HOP_BY_HOP
+ *  - ipIPv6_EXT_HEADER_ROUTING_HEADER
+ *  - ipIPv6_EXT_HEADER_FRAGMENT_HEADER
+ *  - ipIPv6_EXT_HEADER_SECURE_PAYLOAD
+ *  - ipIPv6_EXT_HEADER_AUTHEN_HEADER
+ *  - ipIPv6_EXT_HEADER_DESTINATION_OPTIONS
+ *  - ipIPv6_EXT_HEADER_MOBILITY_HEADER
+ *  - ipPROTOCOL_TCP
+ *     - 1 byte payload
+ */
+void test_eHandleIPv6ExtensionHeaders_TCP_happy_path()
+{
+    eFrameProcessingResult_t eResult;
+    NetworkBufferDescriptor_t * pxNetworkBuffer = prvInitializeNetworkDescriptorWithExtensionHeader( ipPROTOCOL_TCP );
+    TCPHeader_t * pxProtocolHeader = ( TCPHeader_t * ) &( pxNetworkBuffer->pucEthernetBuffer[ ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + 56 ] );
+    uint8_t * pxPayload;
+
+    pxPayload = ( uint8_t * ) ( pxProtocolHeader + 1 );
+    *pxPayload = 'a';
+
+    eResult = eHandleIPv6ExtensionHeaders( pxNetworkBuffer, pdTRUE );
+    TEST_ASSERT_EQUAL( eProcessBuffer, eResult );
+    TEST_ASSERT_EQUAL( ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( TCPHeader_t ) + 1U, pxNetworkBuffer->xDataLength );
+}
+
+/**
+ * @brief test_eHandleIPv6ExtensionHeaders_UDP_happy_path
+ * Prepare a packet have extension with following order. Check if eHandleIPv6ExtensionHeaders determines to process it.
+ *  - ipIPv6_EXT_HEADER_HOP_BY_HOP
+ *  - ipIPv6_EXT_HEADER_ROUTING_HEADER
+ *  - ipIPv6_EXT_HEADER_FRAGMENT_HEADER
+ *  - ipIPv6_EXT_HEADER_SECURE_PAYLOAD
+ *  - ipIPv6_EXT_HEADER_AUTHEN_HEADER
+ *  - ipIPv6_EXT_HEADER_DESTINATION_OPTIONS
+ *  - ipIPv6_EXT_HEADER_MOBILITY_HEADER
+ *  - ipPROTOCOL_UDP
+ *     - 1 byte payload
+ */
+void test_eHandleIPv6ExtensionHeaders_UDP_happy_path()
+{
+    eFrameProcessingResult_t eResult;
+    NetworkBufferDescriptor_t * pxNetworkBuffer = prvInitializeNetworkDescriptorWithExtensionHeader( ipPROTOCOL_UDP );
+    UDPHeader_t * pxProtocolHeader = ( UDPHeader_t * ) &( pxNetworkBuffer->pucEthernetBuffer[ ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + 56 ] );
```

Comment:
Can we define 56

Reply:
Updated. Define a TEST_IPv6_EXTESION_HEADER_LENGTH as one extension header length, then multiply it with the number of headers.